### PR TITLE
[RFC/WIP] Add shopware store api support

### PR DESCRIPTION
**Pull request by m3gg3r**
A first draft of installing shopware store plugins directly from store.

This implementation is not ready for merging at the moment, but it would be great to see some feedback.

TODOs:
- Move the store communication parts to an own service
- Thing about an solution about the 'provider_parameters' because the version is needed to download the right zip
- Add examples
- Add more error handling

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,6 +5,7 @@ This is a CLI based plugin manager that can simply install, uninstall, enable or
 The shopware instance needs not to be fully working for this plugin manager
 to work as it is designed to be independent from the Shopware core itself.
 
+The `StoreApiProvider` is heavily inspired `store-plugin-installer` from `shyim` see [here](https://github.com/shyim/store-plugin-installer)
```
Comment (michz) [inline README.md]:
```suggestion
The `StoreApiProvider` is heavily inspired by `store-plugin-installer` from `shyim` see [here](https://github.com/shyim/store-plugin-installer)
```

File: spec/Provider/StoreApiProviderSpec.php [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+                        'version'  => '0.0.2'
+                    ]
+                ]
+            );
+//        $this->loadFile(
+//            [
+//                'pluginId' => 'awesomePlugin',
+//                'version'  => '0.0.2'
+//            ]
+//        );
```
Comment (michz) [inline spec/Provider/StoreApiProviderSpec.php]: ?

File: src/Service/StreamTranslator.php [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+ * @copyright 2018 solutionDrive GmbH
+ */
+
+namespace sd\SwPluginManager\Service;
+
+use Psr\Http\Message\StreamInterface;
+
+class StreamTranslator implements StreamTranslatorInterface
+{
+    public function translateToArray(StreamInterface $stream)
```
Comment (michz) [inline src/Service/StreamTranslator.php]:
```suggestion
    /**
     * @return object|mixed
     */
    public function translateToArray(StreamInterface $stream)
```

?

File: spec/Provider/StoreApiProviderSpec.php (hunk 2 above)
Comment (m3gg3r) [reply spec/Provider/StoreApiProviderSpec.php]: Done!

File: src/Service/StreamTranslator.php (hunk 3 above)
Comment (m3gg3r) [reply src/Service/StreamTranslator.php]: Done!

**Pull request merged**